packages/@uppy/tus/src/index.ts

```diff
@@ -433,7 +431,11 @@ export default class Tus<M extends Meta, B extends Body> extends BasePlugin<
           // Send along all fields by default.
         : Object.keys(file.meta)
       allowedMetaFields.forEach((item) => {
-        meta[item] = file.meta[item] as string
+        // tus type definition for metadata only accepts `Record<string, string>`
+        // but in reality (at runtime) it accepts `Record<string, unknown>`
+        // tus internally converts everything into a string, but let's do it here instead to be explicit.
+        // because Uppy can have anything inside meta values, (for example relativePath: null is often sent by uppy)
+        meta[item] = String(file.meta[item])
```

Comment:
Given the comment, I don't understand why we'd chose a runtime thing (`String()`) instead of a type thing `as string` if that doesn't matter. To me, it seems the type casting is much preferable.

Reply:
because by using String() we're explicitly saying that we want to convert the value to its string representation (for use in tus), instead of a hidden internal conversion that may or may not happen inside tus (which also breaks tus's type contract)

Reply:
Then the comment is misleading, tus expects `Record<string, string>` and does not convert into a string (it was simply coerced to string, but that's not tus doing it, that's just how JS works)

Reply:
tus does explicitly convert it into a string here using the String() function: https://github.com/tus/tus-js-client/blob/9182f7481d8ca2b5cc7d50e1cdc8b34fd17a171e/lib/upload.js#L920

as far as i can see it does happily accept anything.

what would you write instead?